### PR TITLE
feat: filter flags by username

### PR DESCRIPTION
**Pull request by gasoved**
NodeBB/NodeBB#8489

### CODE REVIEW

File: less/bootstrap/dropdowns.less [hunk 1]
```diff
@@ -195,6 +196,12 @@
   }
 }
 
+.caret-up {
+  .fa-caret-down::before {
+    content: @fa-var-caret-up;
+  }
+}
+
```
Comment (julianlam) [inline less/bootstrap/dropdowns.less:204]: Do we need to modify `dropdowns.less`? Best to leave the vendor file alone if possible.

File: public/persona.js [hunk 2]
```diff
@@ -472,4 +475,24 @@ $(document).ready(function () {
...
+			return;
+		}
+		el.setAttribute('data-text-variant', el.textContent);
+		for (const child of el.childNodes) {
+			if (child.textContent === el.textContent) {
+				child.textContent = textVariant;
+				break;
+			}
+		}
+	}
```
Comment (julianlam) [inline public/persona.js:497]: Perhaps we can come up with a more elegant solution than this... not sure what this does?

Comment (julianlam) [inline public/persona.js:497]: Perhaps we can come up with a more elegant solution than this... not sure what this does?

File: less/bootstrap/dropdowns.less (hunk 1 above)
Comment (julianlam) [inline less/bootstrap/dropdowns.less:204]: Do we need to modify dropdowns.less? Best to leave the vendor file alone if possible.

**Review by julianlam — COMMENTED** [review]
Some concerns I'd like context for, thanks. Otherwise, looks good.

**Comment by julianlam** [comment]
Personally, I think this amount of added code just to change a carat to an upwards carat is overkill. I'd recommend replacing the `<span class="carat"></span>` with `<i class="fa fa-sort"></i>` instead.

File: less/bootstrap/dropdowns.less (hunk 1 above)
Comment (gasoved) [reply less/bootstrap/dropdowns.less:204]: Oh, I hadn't thought it as a vendor file. But also hadn't felt quite right. We can put that style any place you see apt.

**Review by gasoved — COMMENTED** [review]
I admit I sneaked this caret up/down and text-variant thing under this update. :)

This has two purpose. Caret handling is the small one. I was willing to have a general "data-text-variant" attribute handler for awhile. So started from adding it into a theme, hoping to get a suggestion to move it into a more "general" place in core (with some modifications). The "for" loop is the ugly part, but it is there to handle "all" cases. And does not have a performance impact actually, as it might appear to be.

So, I think we shouldn't show the text "More Filters" when that panel is expanded. A small polishment attempt while I was at it. "fa-caret" is the small part, which we can have too if we leave text-variant thing there. Reflect the intent better than the fa-sort, another small polishment.

Sorry about the intense bikeshedding. Ready to nuke after your confirmation.

File: less/bootstrap/dropdowns.less (hunk 1 above)
Comment (julianlam) [reply less/bootstrap/dropdowns.less:204]: Let's remove the use of `<span class="carat"></span>` in favour of `<i class="fa fa-sort"></i>` instead.

**Review by julianlam — CHANGES_REQUESTED** [review]
See slack for comments re: text-variant changes.

tl;dr move it out of theme and into `public/src/flags/list.js[enableFilterForm()]`
